src/NuGet.Core/NuGet.Protocol/Resources/PackageUpdateResource.cs

```diff
@@ -181,31 +199,26 @@ public Uri SourceUri
                         Strings.DefaultSymbolServer));
                 }
 
-                var skipDuplicate = false;
-                await PushPackage(symbolPackagePath, source, apiKey, noServiceEndpoint, skipDuplicate, requestTimeout, log, token, isSnupkgPush: isSymbolEndpointSnupkgCapable);
+                await PushAll(source, apiKey, noServiceEndpoint, skipDuplicate, requestTimeout, log, symbolsToPush, token);
             }
         }
 
-        /// <summary>
-        /// Get the symbols package from the original package. Removes the .nupkg and adds .symbols.nupkg
-        /// </summary>
-        private static string GetSymbolsPath(string packagePath, bool isSnupkg)
-        {
-            var symbolPath = Path.GetFileNameWithoutExtension(packagePath) + (isSnupkg ? NuGetConstants.SnupkgExtension : NuGetConstants.SymbolsExtension);
-            var packageDir = Path.GetDirectoryName(packagePath);
-            return Path.Combine(packageDir, symbolPath);
-        }
-
         private async Task PushPackage(string packagePath,
             string source,
             string apiKey,
             bool noServiceEndpoint,
             bool skipDuplicate,
             TimeSpan requestTimeout,
             ILogger log,
-            CancellationToken token,
-            bool isSnupkgPush)
+            CancellationToken token)
         {
+            var nupkgsToPush = LocalFolderUtility.ResolvePackageFromPath(packagePath, isSnupkg: false);
+
+            if (!LocalFolderUtility.ResolvedAnyPackagePath(nupkgsToPush))
+            {
+                LocalFolderUtility.ErrorFileNotFound(packagePath);
+            }
```

Comment:
This can be replaced by a call to `LocalFolderUtility.EnsurePackageFileExists`

Reply:
There's a subtle difference: I need the collection package paths so that I can [pass them later](https://github.com/NuGet/NuGet.Client/blob/c38da974f2730833901ebb355b7035cf48a1bcbd/src/NuGet.Core/NuGet.Protocol/Resources/PackageUpdateResource.cs#L231). That's why I created helpers.